### ISSUE TITLE
v2.4.1 replaces spaces in filenames with plus signs

### ISSUE THREAD

**Issue by jtwile2** [jfrog/bamboo-artifactory-plugin#51]
Using v2.4.1 of the plugin:

When using either legacy patterns or File Specs, if a filename found by the discovery pattern has spaces in the filename, those spaces are replaces with plus signs.  Log below:

```
build	30-Jan-2018 13:55:36	drwxr-xr-x.    2 root root  4096 Jan 30 13:27 .
build	30-Jan-2018 13:55:36	drwxr-xr-x. 2438 root root 94208 Jan 30 13:50 ..
build	30-Jan-2018 13:55:36	-rw-r--r--.    1 root root     9 Jan 30 13:55 File with spaces.pkg
simple	30-Jan-2018 13:55:36	Finished task 'Make file with spaces' with result: Success
simple	30-Jan-2018 13:55:36	Starting task 'Deploy file' of type 'org.jfrog.bamboo.bamboo-artifactory-plugin:artifactoryGenericTask'
simple	30-Jan-2018 13:55:36	Bamboo Artifactory Plugin version: 2.4.1
error	30-Jan-2018 13:55:36	Agent machine time is lower than the server machine time, please synchronize them.
simple	30-Jan-2018 13:55:36	For pattern: *.pkg 1 artifacts were found.
simple	30-Jan-2018 13:55:36	Deploying artifact: https://artifactory.redacted.build/artifactory/libs-snapshot-local/test-uploads/test/File+with+spaces.pkg
simple	30-Jan-2018 13:55:36	Deploying build descriptor to: https://artifactory.redacted.build/artifactory/api/build
simple	30-Jan-2018 13:55:36	Build successfully deployed. Browse it in Artifactory under https://artifactory.redacted.build/artifactory/webapp/builds/DevOps%20-%20test%20-%20artifact%20upload/6
```

I would expect the artifact to be uploaded as 'File with spaces.pkg' instead of 'File+with+spaces.pkg'

Thanks

**Comment by eyalbe4** [comment]
Thanks for reporting this @jtwile2.
We're looking into this and we'll update soon.

**Comment by yahavi** [comment]
Thanks, @jtwile2.

We opened [BI-444](https://www.jfrog.com/jira/browse/BI-444) to track this issue.

**Comment by yahavi** [comment]
@jtwile2,
We fixed this issue.
Here is a link to the commit: https://github.com/JFrogDev/build-info/commit/2507ad7e1b5bcd5f38fc139096f98f8f8b107f75

Here is a snapshot which resolve this issue: 
https://oss.jfrog.org/oss-snapshot-local/org/jfrog/bamboo/bamboo-artifactory-plugin/2.4.x-SNAPSHOT/bamboo-artifactory-plugin-2.4.x-20180214.160028-5.jar

We'll appreciate you feedback for that!

**Comment by yahavi** [comment]
@jtwile2,
following my last message, please notice that the snapshot is compatible with bamboo 6.2.5 and above.

**Comment by jtwile2** [comment]
@yahavi This is working as expected, thankyou.  Files with plus signs and files with spaces in the filename are deployed correctly.  When is the next release of the plugin expected?  I would prefer to run production with a released version instead of the snapshot.

**Comment by yahavi** [comment]
@jtwile2,
Thanks for your feedback!
The next Bamboo plugin is planned to be released in the next few weeks.

**Comment by bb-tle** [comment]
This same issue also happens with Jenkins Artifactory plugin (v2.14). I wonder if there has already been a fix for it.
Thank you.

**Comment by yahavi** [comment]
Hello @bb-tle, 
The latest release of Jenkins Artifactory plugin (v2.15) contains the fix.

**Comment by bb-tle** [comment]
Thank you @yahavi v2.15 fixed the problem.

**Comment by karapuz50** [comment]
Hello @bb-tle,
we are running Artifactory Version 6.5.13. When uploading helm chart with '+' sign in its name the uploaded artifact got ' ' (space ) instead of '+'.
$ ls file-parser-5.5.1+*
file-parser-5.5.1+1.tgz  file-parser-5.5.1+2.tgz
$ 
On Artifactory:
$ curl -X GET https://++++++++++443/artifactory/api/storage/va-helm/file-parser
{
  "repo" : "va-helm",
  "path" : "/file-parser",
  "created" : "2019-07-21T14:12:03.550Z",
  "createdBy" : "k8s",
  "lastModified" : "2019-07-21T14:12:03.550Z",
  "modifiedBy" : "k8s",
  "lastUpdated" : "2019-07-21T14:12:03.550Z",
  "children" : [ {
    "uri" : "/file-parser-5.5.0.tgz",
    "folder" : false
  }, {
    "uri" : "/file-parser-5.5.1 1-space.tgz",
    "folder" : false
  }, {
    "uri" : "/file-parser-5.5.1 1.tgz",
    "folder" : false
  }, {
    "uri" : "/file-parser-5.5.1 2.tgz",
    "folder" : false
  }, {
    "uri" : "/file-parser-5.5.1-1.tgz",
    "folder" : false
  }, {
    "uri" : "/file-parser-5.5.1-2.tgz",
    "folder" : false
  }, {
    "uri" : "/file-parser-5.5.VERTHREE-13406-SNAPSHOT.tgz",
    "folder" : false
  } ],
  "uri" : "https://artifactory.voyagerlabs.co:443/artifactory/api/storage/va-helm/file-parser"
$

Could you check please if this is a bug?
regards,
Dmitry

**Comment by eyalbe4** [comment]
@karapuz50,
Is your question related to the Bamboo Artifactory Plugin?
If so, can you please be more specific on how you upload the helm chart using Bamboo?

**Comment by karapuz50** [comment]
@eyalbe4 
No, it's not related to any plugin. I've uploaded the artifacts via UI.

**Comment by shuvadipc** [comment]
Please use our newer bamboo-jfrog-plugin for getting more features and support.
